### ISSUE TITLE
Create delete account page

### ISSUE THREAD

**Issue by knjk04** [Project-Books/book-project#678]
**Describe the solution you'd like**
![image](https://user-images.githubusercontent.com/11173328/114820003-3db7da80-9db6-11eb-8f7b-85c59db4629f.png)
Add a red 'Delete account' link on the settings page.

![image](https://user-images.githubusercontent.com/11173328/114819906-19f49480-9db6-11eb-8001-2c774b6c326d.png)

Also, create the delete account page as shown above. This should be responsive. 

The confirm password field should use our existing password component.

Both the back and cancel button should take the user back to the settings page.

For the time being, the 'Export account data' and 'Delete my account' buttons should not do anything.

**Additional context**

Create your branch from the latest `0.2.0` branch on this repository and set the destination of your feature branch in your pull request to `0.2.0`.

**Comment by skeyas** [comment]
Could I work on this?

**Comment by knjk04** [comment]
@skeyas Sure, thanks!

**Comment by knjk04** [comment]
How are you getting on with this?

**Comment by knjk04** [comment]
@skeyas As we have not heard back from you in a while, I will unassign you from the issue. This is in line with our contributing guidelines

**Comment by RivaD2** [comment]
@knjk04 I would be happy to take this one on!

**Comment by knjk04** [comment]
@RivaD2 Thanks!

**Comment by knjk04** [comment]
@RivaD2 We've merged in the endpoint (see https://github.com/Project-Books/book-project/pull/668), so you may want to pull from origin

**Comment by RivaD2** [comment]
@knjk04 What I'm thinking for the approach is:
   - Create a new route for delete account page on `route.tsx` 
   - Add delete account link on `Settings.tsx`
      - Use `Link` with included path to take user to the delete account page once link is clicked on 
     - For building the page itself, I can make use of a container div with a drop shadow and reuse other components we have and use  the optional parameter of 'centered' from the `Layout' component to help me with formatting (and also refer back to older styles to assist with drop shadow etc.). 
   - Regarding the  back and cancel button taking the user back to the settings page,I could either use `Link` again for both, or use the built in history object's method of 'goBack' for the back button and `Link` for the cancel button.

**Comment by knjk04** [comment]
@RivaD2 Your approach sounds good! This page is a bit different to the others, which follow a more consistent format. I think it would be better to avoid using the layout component for this. If we find that we need a similar page to this delete account page in the future, then we can perhaps either modify the layout component to make it more general or create a new shared parent component.

Feel free to create a new issue to make the layout component mobile-friendly. From memory, I thought it was responsive, but I could be wrong

**Comment by RivaD2** [comment]
@knjk04 Sounds good, thank you! 
  Regarding page responsiveness, the nav bar is there even when the page hits the breakpoint. To free up space for mobile, it would have to collapse into another type of menu or we might need to add responsive styles to it separately. I looked at our mock ups but know that we might not have mobile designs quite yet.
 
I am happy to make a ticket for the layout component for additional styling (as it is somewhat responsive but is interconnected tot he nav bar, is styled with margins etc . to work with the nav bar). 
  I can also make a ticket for mobile friendly nav bar too. I just want to confirm this sounds good with you first. Thank you for your time!

**Comment by knjk04** [comment]
@ofvoler What are your thoughts on this?

**Comment by ofvoler** [comment]
I think adding the 'back' in the layout component would be great! I actually have that in some of the mockups but didn't actually implement it. 

Also, I agree that the menu doesn't work on smaller screens. I think that a collapsible hamburger menu or something similar would be great here. It might be good to implement this on vertical tablet screens as well.

TLDR; I think it's great to go ahead with everything that Riva suggested!

**Comment by RivaD2** [comment]
@knjk04 and @ofvoler How would we like to proceed with responsive designs for tickets? In this case, I can try and get the delete Account page as responsive as possible and of course add the `back` arrow into `Layout` along with given props, but until the Nav bar is made responsive, this page (and others) will not be truly responsive. Thank you for your help :)

**Comment by knjk04** [comment]
@RivaD2 We don't have responsive designs in place yet, but we may need to redesign things to make it work well as a progressive web app.

> In this case, should I try and get the delete Account page as responsive as possible now, or should I just compete the ticket as shown above/ add the back arrow into Layout without additional media queries etc.?

Either's fine

> I'm thinking maybe holding off on responsive styles here might be a good idea as I'm not having success styling in a responsive way with the nav bar showing. Does this sound like a good plan?

Sure, that's fine

Thanks!

**Comment by RivaD2** [comment]
This has been merged
